src/SmartComponents/BaselinesPage/EditBaseline/redux/reducers.js

```diff
@@ -93,6 +92,36 @@ export function editBaselineReducer(state = initialState, action) {
                 isCategory: action.payload.isCategory,
                 isSubFact: action.payload.isSubFact
             };
+        case `${types.SELECT_FACT}`: {
+            newEditBaselineTableData = [ ...state.editBaselineTableData ];
+            let isAllSelected = true;
+
+            newEditBaselineTableData.map(row => {
+                if (action.payload.ids.includes(row[0])) {
+                    row.selected = action.payload.isSelected;
+                }
+
+                if (Array.isArray(row[2])) {
+                    row[2].map(subFact => {
+                        if (action.payload.ids.includes(subFact[0])) {
+                            subFact.selected = action.payload.isSelected;
+                        }
+                    });
+                }
+            });
+
+            newEditBaselineTableData.forEach(function(row) {
+                if (!row.selected) {
+                    isAllSelected = false;
```

Comment:
You could put a `break;` after this to exit the loop early since you'll have hit the condition.

Reply:
Can't break or continue in a forEach loop. Can change this to for loop instead?